crates/router/src/db/address.rs

```diff
@@ -91,38 +92,88 @@ impl AddressInterface for Store {
 
 #[async_trait::async_trait]
 impl AddressInterface for MockDb {
+    #[allow(clippy::unwrap_used)]
     async fn find_address(
         &self,
         _address_id: &str,
     ) -> CustomResult<storage::Address, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let addresses = self.addresses.lock().await;
+        Ok(addresses
+            .iter()
+            .find(|address| address.address_id == _address_id)
+            .unwrap()
```

Comment:
Please update your implementation to remove all `unwrap()`s and return suitable errors instead. Although the primary use case of `MockDb` is for testing, we'd prefer to `unwrap()` only in tests and not in the underlying implementation.